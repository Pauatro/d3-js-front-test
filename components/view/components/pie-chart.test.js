import createViewContainer from './view-container.js'
import createPieChart from './pie-chart.js'


it('renders correctly', () => {

    const container = document.createElement('div')
    container.style = "width: 300px; height: 200px;"

    document.body.append(container)

    const {
        svg,
        containerHeight
    } = createViewContainer(container, 0)

    const outerRadius = containerHeight/3.5
    const innerRadius = outerRadius*0.9

    const grouped = [
        {
            "group": "tablet",
            "perc": 60,
            "total": 120000
        },
        {
            "group": "smartphone",
            "perc": 40,
            "total": 80000
        }
    ]

    createPieChart(svg, innerRadius, outerRadius,containerHeight, grouped,0)
    expect(svg).toMatchSnapshot();
});