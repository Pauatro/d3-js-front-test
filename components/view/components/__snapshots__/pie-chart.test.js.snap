// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Selection {
  "_groups": Array [
    Array [
      <svg
        viewBox="0,0,0,0"
      >
        <g
          transform="translate(0,0)"
        >
          <path
            d="M0,0Z"
            fill="#3a6815"
          />
          <path
            d="M0,0Z"
            fill="#93d55d"
          />
          <path
            d="M0,0Z"
            fill="#b3b3b3"
          />
          <path
            d="M0,0Z"
            fill="white"
          />
          <path
            d="M0,0Z"
            fill="white"
          />
          <path
            d="M0,0Z"
            fill="white"
          />
          <path
            d="M0,0Z"
            fill="white"
          />
          <path
            d="M0,0Z"
            fill="white"
          />
        </g>
      </svg>,
    ],
  ],
  "_parents": Array [
    null,
  ],
}
`;
